File: test-automation-core/src/main/java/com/nortal/test/core/rest/interceptors/HeaderInterceptor.java
package com.nortal.test.core.rest.interceptors;

import java.io.IOException;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Component;
import com.nortal.test.core.services.ScenarioContainer;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

@Component
public class HeaderInterceptor implements Interceptor {
	private final ScenarioContainer scenarioContainer;

	public HeaderInterceptor(@Lazy final ScenarioContainer scenarioContainer) {
		this.scenarioContainer = scenarioContainer;
	}

	@Override
	public Response intercept(final Interceptor.Chain chain) throws IOException {

		if (!scenarioContainer.get().getCommon().getHeaders().isEmpty()) {
			return setHeaderValues(chain);
		}

		return chain.proceed(chain.request());
	}

	private Response setHeaderValues(final Chain chain) throws IOException {
		final Request.Builder request = chain.request().newBuilder();
		scenarioContainer.get().getCommon().getHeaders().forEach(request::header);

		return chain.proceed(request.build());
	}
}

File: test-automation-core-old/src/main/java/com/nortal/test/core/rest/interceptors/HeaderInterceptor.kt
package com.nortal.test.core.rest.interceptors

import com.nortal.test.core.services.ScenarioContainer
import com.nortal.test.core.services.ScenarioContainer.get
import okhttp3.Interceptor
import okhttp3.Response
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class HeaderInterceptor(@param:Lazy private val scenarioContainer: ScenarioContainer) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return if (!scenarioContainer.get().common.headers.isEmpty()) {
            setHeaderValues(chain)
        } else chain.proceed(chain.request())
    }

    @Throws(IOException::class)
    private fun setHeaderValues(chain: Interceptor.Chain): Response {
        val request = chain.request().newBuilder()
        scenarioContainer.get().common.headers.forEach { (name: String?, value: String?) ->
            request.header(
                name!!, value!!
            )
        }
        return chain.proceed(request.build())
    }
}